Mathlib/Init/Algebra/Order.lean

```diff
@@ -232,6 +232,9 @@ class LinearOrder (α : Type u) extends PartialOrder α, Min α, Max α :=
   min_def : ∀ a b, min a b = if a ≤ b then a else b := by intros; rfl
   /-- The minimum function is equivalent to the one you get from `maxOfLe`. -/
   max_def : ∀ a b, max a b = if a ≤ b then b else a := by intros; rfl
+  compare a b := compareOfLessAndEq a b
+  /-- Comparison via `compare` is equal to the canonical comparison given decidable `<` and `=`. -/
+  compare_eq_compareOfLessAndEq : ∀ a b, compare a b = compareOfLessAndEq a b := by intros; rfl
```

Comment:
```suggestion
  compare_eq_compareOfLessAndEq (a b) : compare a b = compareOfLessAndEq a b := by rfl
```
(untested)

Reply:
Weirdly, seems like the autoparam still attaches to the overall term, not the inner term, so it still needs `intros`.